src/sas/qtgui/Perspectives/Fitting/FittingWidget.py

```diff
@@ -1658,6 +1673,72 @@ def onSelectCategory(self) -> None:
         self.cbModel.addItems(sorted(models_to_show))
         self.cbModel.blockSignals(False)
 
+    def onInView(self):
+        """
+        Trigers the display of 'InView'
+        """
+        if self._in_view_widget is None:
+            self._in_view_widget = InViewWidget(parent=self)
+            self._in_view_widget.destroyed.connect(lambda: setattr(self, '_in_view_widget', None))
+
+            # When InView is closed, re-enable editing and clear reference
+            def _on_inview_closed():
+                self._in_view_widget = None
+                # Enabling paramters edits
+                self.lstParams.setEnabled(True)
+                self.cbCategory.setEnabled(True)
+                self.cmdFit.setEnabled(True)
+                self.cmdPlot.setEnabled(True)
+
+
+            self._in_view_widget.destroyed.connect(_on_inview_closed)
+
+        self.lstParams.setEnabled(False)
+        self.cbCategory.setEnabled(False)
+        self.cmdFit.setEnabled(False)
+        self.cmdPlot.setEnabled(False)
+
+        ###
+        # Validate parameter bounds for InView (no +/-inf allowed)
+        params = list(self.main_params_to_fit)
+        if self.chkPolydispersity.isChecked():
+            params += list(self.polydispersity_widget.poly_params_to_fit)
+        has_infinite_bounds = False
+        for p in params:
+            try:
+                details = self.logic.kernel_module.details.get(p)
+                if details is None or len(details) < 3:
+                    continue
+                pmin, pmax = details[1], details[2]
+                # Disallow non-finite bounds
+                if not np.isfinite(pmin) or not np.isfinite(pmax):
+                    has_infinite_bounds = True
+                    break
+            except Exception:
+                # If anything odd, be conservative and flag
+                has_infinite_bounds = True
+                break
+        if has_infinite_bounds:
+            QtWidgets.QMessageBox.warning(self, "InView",
+                "Some parameters have infinite bounds. Please set appropriate min/max ranges.")
+            # Re-enable controls since we are not opening InView
+            self.lstParams.setEnabled(True)
+            self.cbCategory.setEnabled(True)
+            self.cmdFit.setEnabled(True)
+            self.cmdPlot.setEnabled(True)
+            return
```

Comment:
Simplification:

```suggestion
                if not np.isfinite(pmin) or not np.isfinite(pmax):
                    # Fall through to exception handling, which mimics behavior required here.
                    raise Exception()
            except Exception:
                # If anything odd, be conservative and flag
                QtWidgets.QMessageBox.warning(self, "InView",
                "Some parameters have infinite bounds. Please set appropriate min/max ranges.")
                # Re-enable controls since we are not opening InView
                self._on_inview_closed()
                return
```